src/sagemaker/local/image.py

```diff
@@ -102,7 +106,12 @@ def train(self, input_data_config, hyperparameters):
             self.write_config_files(host, hyperparameters, input_data_config)
             shutil.copytree(data_dir, os.path.join(self.container_root, host, 'input', 'data'))
 
-        compose_data = self._generate_compose_file('train', additional_volumes=volumes)
+        training_env_vars = {
+            REGION_ENV_NAME: self.sagemaker_session.boto_session.region_name,
```

Comment:
there are cases in which boto_session can be None. I would add an if guard  here.